File: Api2.0/src/main/java/com/justanothervitor/api_2/models/payloads/request/NotesFilterRequest.java
package com.justanothervitor.api_2.models.payloads.request;

import lombok.Data;

import java.math.BigDecimal;

@Data
public class NotesFilterRequest {
    private String searchContent;
    private Long categoryId;
    private BigDecimal minPrice;
    private BigDecimal maxPrice;

    private int page = 0;
    private int size = 10;

    private String orderFor = "name";
    private String direction = "asc";
}

File: web-front/src/app/Models/notesFilterRequest.ts
/**
 * Payload para pesquisa paginada de anotações.
 * Espelha o NotesFilterRequest.java do backend.
 */
export interface NotesFilterRequest {
  searchContent?: string;
  categoryId?: number;
  minPrice?: number;
  maxPrice?: number;
  page: number;
  size: number;
  orderFor?: string;
  direction?: string;
}
